### DISCUSSION TITLE
.workspace_special not working and switching between workspaces only.

### DISCUSSION THREAD

**Discussion by kozlekkinski** [H3rmt/hyprshell#83] [Q&A] [answered]
Hi

1)  .workspace_special isn't working.  I have this in my CSS (the CSS itself works, as some other customizations I've made show up): `.workspace_special {
    border: 3px solid rgba(0, 255, 0, 0.4);
}`
But it doesn't show any of the active workspaces with negative IDs (be it scratchpad or not).
![image](https://github.com/user-attachments/assets/071be79f-2bd9-49fa-8640-ea6072e25d9b)

2) Is it possible to swap between workspaces only? I don't want to have to choose between 2 windows of the same workspace, or have to tab more times for "no reason" (at least for me).

Thanks

**Comment by H3rmt** [comment, marked as answer]
> 2. Is it possible to swap between workspaces only? I don't want to have to choose between 2 windows of the same workspace, or have to tab more times for "no reason" (at least for me).

You can use the `--switch-type` to switch between clients / workspaces / monitors.

Pass this flag to the `hyprswitch gui .... ` call with `workspace` as it's value

**Comment by H3rmt** [comment]
> 1. workspace_special isn't working. I have this in my CSS (the CSS itself works, as some other customizations I've made show up): .workspace_special { border: 3px solid rgba(0, 255, 0, 0.4); }
But it doesn't show any of the active workspaces with negative IDs (be it scratchpad or not).

Use the `--include-special-workspaces` flag (pass it to `hyprswitch gui`) to include special workspaces. (They are excluded by default)

There is a problem with special workspaces, they can only be toggled.
This can cause hyprswitch to get confused whether the special workspace is opened or not which can cause issues when switching to clients on special workspaces.
I plan to fix this some time in the future.

**Reply by kozlekkinski** [reply]
Thank you for both your answers, I didn't realize i could use the --help flag on the options of hyprswitch too, and i only referred to the README.md, so I missed those.
On another note, I think the `--sort-recent` flag doesn't work with `--switch-type workspace` (by that i mean that it doesn't sort anything, it's just in ascending number).

**Reply by H3rmt** [reply]
`--sort-recent` only works with clients (windows) as the focus history is provided by Hyprland. There is (current) no similar information for workspaces or monitors
